web/chat/markdown.react.js

```diff
@@ -0,0 +1,33 @@
+// @flow
+
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import * as SimpleMarkdown from 'simple-markdown';
+
+import rules from './rules.react';
+
+type Props = {|
+  children: string,
+|};
+class Markdown extends React.PureComponent<Props> {
+  static propTypes = {
+    children: PropTypes.string.isRequired,
+  };
+
+  ast: SimpleMarkdown.Parser;
+  output: SimpleMarkdown.ReactOutput;
+
+  constructor(props: Props) {
+    super(props);
+
+    const parser = SimpleMarkdown.parserFor(rules());
+    this.ast = parser(this.props.children, { inline: true });
+    this.output = SimpleMarkdown.outputFor(rules(), 'react');
+  }
```

Comment:
I wonder if we should handle the emoji logic here, like we're doing on native

Reply:
I think we shouldn't change this on web, but since we got rid of state.linkPresent, we might also do the same with state.emojiOnly, and handle this like before in `InnerTextMessage` on native, like before.